### PR TITLE
Base for wp sites monitoring

### PR DESCRIPTION
**Pull request by GregLeBarbar**
This pull request lay the foundation for monitoring our wordpresses. It's based on prometheus-wwp.epfl.ch which collect all the metrics.

It includes:

  - prometheus-wwp is now acessible to the world behind a basic authentification (welcome traefik proxy in the pod) (c83e85c8).

  - Improved roles for `ansible -t wp.backup task`: when this task in ran, it push data to the pushgateway (538a43de9bf4a694a8fbb578ce198080f7310c08) . Metrics are named `restic_*` (c83e85c8).

  - The "prober" container has been improved and provide additional data to the wp_menus:
  - its query URL is now `/wpprobe` instead of `/wpmenus` (0b3ef1641a131f4e65a6d4a1cc4550f3f6ffce6f)
  - new gauges for site languages (d11b083d66c21a2ad7ab3ad0c4edb7d56ff319d7)
  - new metrics for counting languages avalaible in a site (538a43de9bf4a694a8fbb578ce198080f7310c08)
  - new metrics for pages (c462c0dd010ed43f4c7e32cbddaad6601149968e)
  - logs ouput are now clean and readable.

Please note that "unmanaged" websites are exclued (c462c0dd010ed43f4c7e32cbddaad6601149968e).

More globally, we've tried to always have the `url` and `wp_env` label in the metrics. Please keep that in mind if adding more metrics in the future. It's useful because of the templating of Grafana dashboards.

### CODE REVIEW

File: ansible/roles/wordpress-instance/tasks/backup.yml [hunk 1]
```diff
@@ -13,70 +14,61 @@
...
+
+      # Move tags forward
       date_short="$(date +%Y%m%d)"
       date_full="$(date +%Y%m%d-%H%M%S)"
       restic -r {{ backup_restic_repo_files }} tag \
         --remove latest --remove "$date_short" --remove "$date_full"
       restic -r {{ backup_restic_repo_sql   }} tag \
         --remove latest --remove "$date_short" --remove "$date_full"
-      restic -r {{ backup_restic_repo_files }} tag {{ snapshot_id.files }} \
+      restic -r {{ backup_restic_repo_files }} tag "$files_snapshot_id" \
```
Comment (domq) [inline ansible/roles/wordpress-instance/tasks/backup.yml:61]: Je ne sais pas exactement pourquoi, mais le résultat de ce changement est qu'il n'y a plus de tags `latest`...

**Review by domq — CHANGES_REQUESTED** [review]

File: ansible/roles/wordpress-instance/tasks/backup.yml (hunk 1 above)
Comment (ponsfrilus) [reply ansible/roles/wordpress-instance/tasks/backup.yml:63]: Fixed by af71410

File: ansible/roles/wordpress-openshift-namespace/vars/monitoring-vars.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
-monitoring_prober_image_name: monitoring-menus-probe
-
+monitoring_prober_image_name: monitoring-wpprobe
 
+_prometheus_secrets: "{{ lookup('file', '/keybase/team/epfl_wp_prod/ansible_prometheus_secrets.yml') | from_yaml }}"
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/vars/monitoring-vars.yml:3]: Il faut mettre ces secrets sous Git — Cf. Les usages pré-existants de `| eyaml(eyaml_keys)`

**Review by domq — CHANGES_REQUESTED** [review]

File: ansible/roles/wordpress-openshift-namespace/vars/monitoring-vars.yml (hunk 2 above)
Comment (ponsfrilus) [reply ansible/roles/wordpress-openshift-namespace/vars/monitoring-vars.yml]: Please see 9c70989

File: ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    -
+      user: admin
+      pass: ENC[PKCS7,MIIBiQYJKoZIhvcNAQcDoIIBejCCAXYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEARyy4MzUB/2rmOpESvkh6fY0QIi8UsOx7MoJ9Fq6OhEnEyc/cDI0e9zJ8S+ubbP5eSfBi4iqZY6QYL0IX4K/pToyOnNrb5sG8VBlKQ2PYSCXu+ssfRW9O0uJVuDkDUwYVa9RK0yY2CWurEhxrXodHBthVUTiZQIp+1BxZaftKALysTR5HOURWC96tFh1BYfMEKju8mhEXgmeq55Dgx9RanNiS7fcCLZ4uSmlqNajb4RagEergPhoBkYxDbZs7bNXkDjXc6mVlVNqyktDAUFmUJGpMlGNn7kxyoiyB1yc+5hxoJEhaUVmYT0w+HiPELqF33FR0l0oLcz91vkxea20ZCjBMBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBBb3Hi/I/UNylgaqUZiG/ZugCDzps69dZ93iQ0VUulOoyV7+TAJxsx3egmXy60l/WLUFQ==]
+
+prometheus_htpass: "\n{% for cred in prometheus.credentials %}\n          - \"{{ cred.user }}:{{ cred.pass | eyaml(eyaml_keys) | password_hash('bcrypt') }}\"\n{% endfor %}"
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml:29]: Est-ce qu'on arrive à déplacer ça ailleurs ?

File: ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#     --pkcs7-public-key ansible/eyaml/epfl_wp_prod.pem \
+#     ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml
+#
+# 💡 This *will not* decrypt anything for you because chances are, you
+# don't need to decrypt secrets (YA RLY). For those cases where you
+# do, wielding the private key from Keybase is left as an exercise to
+# the reader.
+
+prometheus:
+  # Note that due to padding bits within the encoding, the last character should always be one of [.Oeu]
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml:16]: Selon conversation, il semble que ça pourrait ne pas être nécessaire ?

**Review by domq — APPROVED** [review]

File: ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml (hunk 3 above)
Comment (ponsfrilus) [reply ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml]: Please see 18b784c4376622f9c440b8a418e70a4f6f0cbe4b

File: ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml (hunk 4 above)
Comment (ponsfrilus) [reply ansible/roles/wordpress-openshift-namespace/vars/secrets-prometheus-wwp.yml]: Please see 18b784c4376622f9c440b8a418e70a4f6f0cbe4b

**Pull request merged**